### DISCUSSION TITLE
"Watch On Nebula" button to youtube

### DISCUSSION THREAD

**Discussion by dereineLauch** [cpiber/NebulaEnhance#7] [Ideas]
Often when i  watch a YouTube video, i dont know if the creator is on Nebula. So i go to the [Nebula website](https://nebula.app/) and search for the creator. 
THAT'S TO COMPLICATED.
So i want a button on YouTube that show you if the creator is on Nebula. For example like that (near the sub button):
![image](https://user-images.githubusercontent.com/84781281/131896690-813a7886-bf73-4026-82d7-46ec7b360479.png)

[The same idea had a Reddit user.](https://www.reddit.com/r/watchnebula/comments/lgg34o/feasibility_of_a_browser_plugin_that_adds_a_watch/)
[There is already an extension for that, but only for Firefox.](https://github.com/parkeraddison/watch-on-nebula)

**Comment by cpiber** [comment]
I found that extension as well when I started working on this, didn't realize it was only for Firefox.

It also doesn't look like they are currently working that much on it... Still, I think it'd be rude to just steal their extension basically, so I'd ask you to please open an issue with them regarding chrome support. Edit: I opened one.

Also I'd have the same problems as they have for matching the video itself: parkeraddison/watch-on-nebula#2. Though I already have implemented code to match titles based on IFIDF, so I'd be pretty set up with this I think.

If they aren't ready to work on it however I'd consider adding it, though like the YouTube comments (basically the reverse of this) I'd gate it behind optional permissions.

**Reply by parkeraddison** [reply]
Cool idea to use tfidf for the time being. Hopefully the Nebula team can release a simple API in the future!

**Reply by cpiber** [reply]
I already use it to match the youtube titles, they don't have an api to search in a channel either and the general search api is too expensive anyways.

But nebula actually has a search api (`https://content.watchnebula.com/search/video/?text=<search>`), but I don't think it supports searching in one channel either... Setting a channel always produces 0 results :(

I'm currently on vacation, I'll start working on this in a week!

**Reply by cpiber** [reply]
I now have a working version, could you please try it out? You can download it here: 
[enhancer_for_nebula-1.0.02.zip](https://github.com/cpiber/NebulaEnhance/files/7162322/enhancer_for_nebula-1.0.02.zip)

To install:
- in Chrome, please go to Setting, click Extensions (lower left), enable Developer Mode (upper right) and select "Load unpacked". Here select the unzipped folder (should contain folders _locale, icons, scripts, styles).
- in Firefox, please go to `about:addons`, click on the gear (top right), select Debug addons and click "Load Temporary Addon". Here select the zip you downloaded.
